rhizome/lib/vm_setup.rb

```diff
@@ -173,20 +186,28 @@ def routes4(ip4, ip4_local)
     r "ip -n #{q_vm} addr replace #{vethi}/32 dev vethi#{q_vm}"
     # default?
     r "ip -n #{q_vm} route replace #{vetho} dev vethi#{q_vm}"
-    r "ip -n #{q_vm} route replace #{vm} dev tap#{q_vm}" if ip4
-    r "ip -n #{q_vm} route replace default via #{vetho} dev vethi#{q_vm}"
 
-    r "ip netns exec #{q_vm} bash -c 'echo 1 > /proc/sys/net/ipv4/conf/vethi#{q_vm}/proxy_arp'"
-    r "ip netns exec #{q_vm} bash -c 'echo 1 > /proc/sys/net/ipv4/conf/tap#{q_vm}/proxy_arp'"
+    nics.each do |net6, net4, tapname, mac|
+      r "ip -n #{q_vm} route replace #{vm} dev #{tapname}" if ip4
+      r "ip -n #{q_vm} route replace default via #{vetho} dev vethi#{q_vm}"
+
+      r "ip netns exec #{q_vm} bash -c 'echo 1 > /proc/sys/net/ipv4/conf/vethi#{q_vm}/proxy_arp'"
+      r "ip netns exec #{q_vm} bash -c 'echo 1 > /proc/sys/net/ipv4/conf/#{tapname}/proxy_arp'"
 
-    r "ip -n #{q_vm} addr replace 192.168.0.1/24 dev tap#{q_vm}"
+      r "ip -n #{q_vm} addr replace 192.168.0.1/16 dev #{tapname}"
+      r "ip -n #{q_vm} addr replace 10.0.0.1/8 dev #{tapname}"
+      r "ip -n #{q_vm} addr replace 172.16.0.1/12 dev #{tapname}"
+    end
   end
 
-  def nat4(ip4, private_ipv4)
+  def nat4(ip4, nics)
     return unless ip4
     public_sub = NetAddr::IPv4Net.parse(ip4)
     public_ipv4 = public_sub.network.to_s
 
+    # TODO: this needs to be fixed, the public to private NAT will be an issue??
+    # maybe it's just OK?
```

Comment:
can we resolve this question?

Reply:
It's just OK for now, since we have 1 nic attached atm. So the NAT can be performed from public ip to whatever the first NIC has as a private ip.

Reply:
I think it's even fine when we start supporting multiple nics.